nntrainer/layers/layer_node.cpp

```diff
@@ -139,6 +140,14 @@ int LayerNode::setProperty(std::vector<std::string> properties) {
     std::get<props::Distribute>(*layer_node_props).get();
   auto left_properties = loadProperties(properties, *layer_node_props);
 
+  size_t num_output_tensors = getNumOutputTensors();
+
+  if (num_output_tensors > 1) {
+    init_context =
+      InitLayerContext(init_context.getInputDimensions(),
+                       std::max((unsigned int)num_output_tensors, 1u));
```

Comment:
also resize output_layers to match?
also add check in layer_node.h L380 that when setOutputLayers is called from outside, this will also be resized (maybe check for mismatch)?
maybe create resizeOutputDimensions() like resizeInputDimensions() for help.

Reply:
right. we need to think the mismatch. I will fix and add some more code in next pr which is considering output layers. Meanwhile, we need to think the case.. Multi output connection layers with multi output tensors. In this pr, those are not considered yet.

Reply:
yes.